bakery/checkers/checkers.go

```diff
@@ -245,101 +222,20 @@ func ErrorCaveatf(f string, a ...interface{}) Caveat {
 	return firstParty(CondError, fmt.Sprintf(f, a...))
 }
 
-// AllowCaveat returns a caveat that will deny attempts to use the
-// macaroon to perform any operation other than those listed. Operations
-// must not contain a space.
-func AllowCaveat(op ...string) Caveat {
-	if len(op) == 0 {
-		return ErrorCaveatf("no operations allowed")
-	}
-	return operationCaveat(CondAllow, op)
-}
-
-// DenyCaveat returns a caveat that will deny attempts to use the
-// macaroon to perform any of the listed operations. Operations
-// must not contain a space.
-func DenyCaveat(op ...string) Caveat {
-	return operationCaveat(CondDeny, op)
-}
-
-// operationCaveat is a helper for AllowCaveat and DenyCaveat. It checks
-// that all operation names are valid before createing the caveat.
-func operationCaveat(cond string, op []string) Caveat {
-	for _, o := range op {
-		if strings.IndexByte(o, ' ') != -1 {
-			return ErrorCaveatf("invalid operation name %q", o)
-		}
-	}
-	return firstParty(cond, strings.Join(op, " "))
-}
-
-// OperationsChecker checks any allow or deny caveats
-// with respect to all the named operations in the slice.
-// An allow caveat must allow all the operations in the
-// slice; a deny caveat will fail if it denies any operation in the
-// slice.
-type OperationsChecker []string
-
-// Condition implements Checker.Condition.
-func (OperationsChecker) Condition() string {
-	return ""
-}
+type checkerInfoByName []CheckerInfo
 
-// Check implements Checker.Check.
-func (os OperationsChecker) Check(cond, arg string) error {
-	if len(os) == 0 {
-		switch cond {
-		case CondDeny:
-			return nil
-		case CondAllow:
-			f := strings.Fields(arg)
-			if len(f) == 0 {
-				return errgo.New("no operations allowed")
-			}
-			return errgo.Newf("%s not allowed", f[0])
-		default:
-			return ErrCaveatNotRecognized
-		}
+func (c checkerInfoByName) Less(i, j int) bool {
```

Comment:
I don't think it's needed here. Of the 221 implementations of sort.Interface in the the Go standard library and the packages under golang.org, I see exactly one that has a doc comment - the sort example.